plugin/evm/message/syncable.go

```diff
@@ -14,78 +14,107 @@ import (
 	"github.com/ava-labs/avalanchego/snow/engine/snowman/block"
 )
 
-var _ block.StateSummary = &SyncSummary{}
+var _ Syncable = &BlockSyncSummary{}
 
-// SyncSummary provides the information necessary to sync a node starting
+type Syncable interface {
+	block.StateSummary
+	GetBlockNumber() uint64
+	GetBlockHash() common.Hash
+	GetBlockRoot() common.Hash
+}
+
+type SyncableParser interface {
+	ParseFromBytes(summaryBytes []byte, acceptImpl AcceptImplFn) (Syncable, error)
+}
+
+type AcceptImplFn func(Syncable) (block.StateSyncMode, error)
+
+// BlockSyncSummary provides the information necessary to sync a node starting
 // at the given block.
-type SyncSummary struct {
+type BlockSyncSummary struct {
 	BlockNumber uint64      `serialize:"true"`
 	BlockHash   common.Hash `serialize:"true"`
 	BlockRoot   common.Hash `serialize:"true"`
-	AtomicRoot  common.Hash `serialize:"true"`
 
 	summaryID  ids.ID
 	bytes      []byte
-	acceptImpl func(SyncSummary) (block.StateSyncMode, error)
+	acceptImpl AcceptImplFn
+}
+
+type BlockSyncSummaryParser struct{}
+
+func NewBlockSyncSummaryParser() SyncableParser {
+	return &BlockSyncSummaryParser{}
 }
 
-func NewSyncSummaryFromBytes(summaryBytes []byte, acceptImpl func(SyncSummary) (block.StateSyncMode, error)) (SyncSummary, error) {
-	summary := SyncSummary{}
+func (b *BlockSyncSummaryParser) ParseFromBytes(summaryBytes []byte, acceptImpl AcceptImplFn) (Syncable, error) {
+	summary := BlockSyncSummary{}
 	if codecVersion, err := Codec.Unmarshal(summaryBytes, &summary); err != nil {
-		return SyncSummary{}, err
+		return nil, err
 	} else if codecVersion != Version {
-		return SyncSummary{}, fmt.Errorf("failed to parse syncable summary due to unexpected codec version (%d != %d)", codecVersion, Version)
+		return nil, fmt.Errorf("failed to parse syncable summary due to unexpected codec version (%d != %d)", codecVersion, Version)
 	}
 
 	summary.bytes = summaryBytes
 	summaryID, err := ids.ToID(crypto.Keccak256(summaryBytes))
 	if err != nil {
-		return SyncSummary{}, err
+		return nil, err
 	}
 	summary.summaryID = summaryID
 	summary.acceptImpl = acceptImpl
-	return summary, nil
+	return &summary, nil
 }
 
-func NewSyncSummary(blockHash common.Hash, blockNumber uint64, blockRoot common.Hash, atomicRoot common.Hash) (SyncSummary, error) {
-	summary := SyncSummary{
+func NewBlockSyncSummary(blockHash common.Hash, blockNumber uint64, blockRoot common.Hash) (Syncable, error) {
+	summary := BlockSyncSummary{
 		BlockNumber: blockNumber,
 		BlockHash:   blockHash,
 		BlockRoot:   blockRoot,
-		AtomicRoot:  atomicRoot,
 	}
 	bytes, err := Codec.Marshal(Version, &summary)
 	if err != nil {
-		return SyncSummary{}, err
+		return nil, err
```

Comment:
nit a bit out of scope, but whilst we're at changing this line:

```suggestion
		return nil, fmt.Errorf("encoding block sync summary: %s", err)
```